php_phongo.c

```diff
@@ -3781,16 +3787,24 @@ static HashTable* php_phongo_std_get_gc(phongo_compat_object_handler_type* objec
 /* {{{ PHP_MINIT_FUNCTION */
 PHP_MINIT_FUNCTION(mongodb)
 {
+	bson_mem_vtable_t bson_mem_vtable = {
+		php_phongo_malloc,
+		php_phongo_calloc,
+		php_phongo_realloc,
+		php_phongo_free,
+	};
+
 	(void) type; /* We don't care if we are loaded via dl() or extension= */
 
 	REGISTER_INI_ENTRIES();
 
-	/* Initialize libmongoc */
+	/* Assign our custom vtable to libbson, so all memory allocation in libbson
+	 * (and libmongoc) will use PHP's persistent memory API. After doing so,
+	 * initialize libmongoc. Later, we will shutdown libmongoc and restore
+	 * libbson's vtable in the final GSHUTDOWN. */
+	bson_mem_set_vtable(&bson_mem_vtable);
```

Comment:
`bson_mem_set_vtable` does a struct assignment, which is sufficient to get our function pointers copied into libbson's own internal global (`gMemVtable`). We don't actually need this parameter to stick around as a global.